### PR TITLE
bpo-22859: deprecating unittest.TestProgram.usageExit

### PR DESCRIPTION
**Pull request by carlosdmz**
https://bugs.python.org/issue22859

### CODE REVIEW

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]
As mentioned on the PR, I would suggest making it raise DeprecationWarning first.

**Comment by carlosdamazio** [comment]
@JelleZijlstra Done, `usageExit` is marked as being deprecated and will be removed in 3.13.

File: Misc/NEWS.d/next/Library/2021-12-29-19-37-49.bpo-22859.AixHW7.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+`unittest.TestProgram.usageExit` is marked deprecated, to be removed in 3.13.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2021-12-29-19-37-49.bpo-22859.AixHW7.rst]:
```suggestion
:meth:`~unittest.TestProgram.usageExit` is marked deprecated, to be removed in 3.13.
```

Comment (kumaraditya303) [reply Misc/NEWS.d/next/Library/2021-12-29-19-37-49.bpo-22859.AixHW7.rst]: This will fix the docs CI.

**Comment by carlosdamazio** [comment]
Thank you @kumaraditya303 for fixing the news entry.

File: Lib/unittest/main.py [hunk 2]
```diff
@@ -101,6 +102,8 @@ def __init__(self, module='__main__', defaultTest=None, argv=None,
         self.runTests()
 
     def usageExit(self, msg=None):
+        warnings.warn("function \"usageExit\" is being deprecated and will be"
+                      " removed in 3.13", DeprecationWarning)
```
Comment (JelleZijlstra) [inline Lib/unittest/main.py]:
```suggestion
        warnings.warn("TestProgram.usageExit() is deprecated and will be"
                      " removed in Python 3.13", DeprecationWarning)
```

**Review by kumaraditya303 — APPROVED** [review]

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**